tests/locator_test.go

```diff
@@ -58,16 +58,21 @@ func TestLocatorCheck(t *testing.T) {
 	p := tb.NewPage(nil)
 	require.NotNil(t, p.Goto(tb.staticURL("/locators.html"), nil))
 
-	// Selecting a single element and checking it is OK.
-	t.Run("ok", func(t *testing.T) {
-		check := func() bool {
-			cr := p.Evaluate(tb.toGojaValue(`() => window.check`))
-			return cr.(goja.Value).ToBoolean() //nolint:forcetypeassert
-		}
-		input := p.Locator("#input", nil)
-		input.Check(nil)
+	check := func() bool {
+		cr := p.Evaluate(tb.toGojaValue(`() => window.check`))
+		return cr.(goja.Value).ToBoolean() //nolint:forcetypeassert
+	}
+
+	t.Run("check", func(t *testing.T) {
+		checkbox := p.Locator("#inputCheckbox", nil)
+		checkbox.Check(nil)
 		require.True(t, check(), "could not check the input box")
 	})
+	t.Run("uncheck", func(t *testing.T) {
+		checkbox := p.Locator("#inputCheckbox", nil)
+		checkbox.Uncheck(nil)
+		require.False(t, check(), "could not uncheck the input box")
+	})
```

Comment:
My comment about merging check/uncheck tests was about calling both methods in the same test, one after the other. Otherwise you can't ensure that this test is actually changing the `window.check` state, and that it's not just returning the default `false` value.

Reply:
Hmm, got it. It is similar to what you suggest. In this case, it first checks the element in the first test; since the element is checked, the 2nd test tries to uncheck it. Dependency between tests is not good, so I will change it as you suggested. Thanks for letting me know 🙂

Reply:
And, what you suggested is also better in terms of testing the Locator behavior, so 👍  See: #367

Reply:
> Dependency between tests is not good

Right, and you can't ensure the order they'll run in.